quartz-core/src/main/java/org/quartz/impl/jdbcjobstore/StdJDBCDelegate.java

```diff
@@ -1158,9 +1158,9 @@ public int updateTrigger(Connection conn, OperableTrigger trigger, String state,
             JobDetail jobDetail) throws SQLException, IOException {
 
         // save some clock cycles by unnecessarily writing job data blob ...
-        boolean updateJobData = trigger.getJobDataMap().isDirty();
+        boolean updateJobData = trigger.getJobDataMap().isDirty() && (trigger.getJobDataMap().size() > 0);
```

Comment:
Yeah, good catch! I think just the isDirty() check is enough without the size. I will update the PR when I get a chance. Thanks!